src/v/kafka/server/handlers/topics/topic_utils.h

```diff
@@ -116,23 +117,27 @@ void append_cluster_results(
 // Converts objects representing KafkaAPI message to objects consumed
 // by cluster::controller API
 // clang-format off
-template<typename KafkaApiTypeIter>
+template<typename KafkaApiTypeIter, typename ErrIter>
 requires TopicRequestItem<typename KafkaApiTypeIter::value_type> &&
+         TopicResultIterator<ErrIter> &&
 requires(KafkaApiTypeIter it) {
     to_cluster_type(*it);
 }
 // clang-format on
-auto to_cluster_type(KafkaApiTypeIter begin, KafkaApiTypeIter end)
-  -> chunked_vector<decltype(to_cluster_type(*begin))> {
-    chunked_vector<decltype(to_cluster_type(*begin))> cluster_types;
+auto to_cluster_type(
+  KafkaApiTypeIter begin, KafkaApiTypeIter end, ErrIter out_it)
+  -> chunked_vector<cluster::custom_assignable_topic_configuration> {
+    chunked_vector<cluster::custom_assignable_topic_configuration>
+      cluster_types;
     cluster_types.reserve(std::distance(begin, end));
-    std::transform(
-      begin,
-      end,
-      std::back_inserter(cluster_types),
-      [](const typename KafkaApiTypeIter::value_type& kafka_type) {
-          return to_cluster_type(kafka_type);
-      });
+    for (auto it = begin; it != end; ++it) {
+        try {
+            cluster_types.emplace_back(to_cluster_type(*it));
+        } catch (...) {
+            out_it = generate_error(
+              *it, error_code::invalid_config, "Configuration is invalid");
```

Comment:
This automagically advances the iterator as well under the hood because it is a `std::back_inserter(response.data.topics)`, right? I'm wondering if `ErrIter` needs to be specifically an `insert_iterator` here or if there's a good way to describe why this "just works" without the need to advance `out_it`. Or maybe we should just advance `out_it` here anyway.

I think it's worth adding a test for a partially successful rpk update that would trigger this.

Reply:
`operator=` for `std::back_insert_iterator` acts as a `push_back()` to the underlying container, there is no need to advance the iterator.

See: https://en.cppreference.com/w/cpp/iterator/back_insert_iterator/operator%3D

Reply:
https://godbolt.org/z/TfrqvaWdh

output iterators are auto advancing and they can't be advanced manually (it doesn't compile because std::next, std::advance for output iterators are deleted)

agree that a test with multiple failures would be helpful

Reply:
Fair enough, I didn't know that all stdlib output iterators were auto-advancing. I guess the standard allows custom output iterators, in which case this could still break. But it's unlikely that we'll introduce a custom non-auto-advancing output iterator here.

https://godbolt.org/z/9qMqETP78

Reply:
Reading this concept https://en.cppreference.com/w/cpp/iterator/output_iterator makes me think that the safest/best way is to use

`*out_it++ = generate_error` and we won't have any additional discussions... and no one will wonder what's happening there for more than a second